docs/tutorials/typescript/subscriptions/index.md

```diff
@@ -1,416 +1,417 @@
+  } catch (err: any) {
+    if (err.details) console.error(err.details);
+    else console.error(err);
+  }
+```
+<!--SNIPEND-->
 
-## Queries and Reusable Functions
+You now know how to obtain a handle on a Workflow Execution and send a Signal to it through the Client.
 
-The last requirement is about being able to look up (and change) customer info.
-The complement to Signals is Queries, where we can get data out of a running Workflow, and they have a very similar API:
+### Invoke a Query from the Client
 
+Sending a Query through the Client is also very similar. The SDK provides the `WorkflowHandle.query` method to query a running or completed Workflow, allowing us to see into details of the `subscriptionWorkflow`, whether it is still running or completed.
+
+<!--SNIPSTART subscription-ts-querybillinginfo-query {"selectedLines": ["11-18", "25-34"]}-->
+[src/scripts/query-billinginfo.ts](https://github.com/temporalio/subscription-workflow-project-template-typescript/blob/main/src/scripts/query-billinginfo.ts)
 ```ts
-// not suggested code - just for illustrative purposes
-const amountChargedQuery = wf.defineQuery<number>('amountChargedQuery');
-const updateAmountCharged = wf.defineSignal<number>('updateAmountCharged');
-
-// inside Workflow
-let amountCharged = customer.initialBillingPeriodCharge;
-wf.setHandler(amountChargedQuery, () => amountCharged);
-wf.setHandler(
-  updateAmountCharged,
-  (newAmount: number) => void (amountCharged = newAmount)
-);
-// do stuff with amountCharged
+// ...
+  const subscriptionWorkflowExecution = await client.workflow.start(
+    subscriptionWorkflow,
+    {
+      args: [customer],
+      taskQueue: TASK_QUEUE_NAME,
+      workflowId: `subscription-${customer.id}`,
+    }
+  );
+// ...
+      const billingPeriodNumber =
+        await subscriptionWorkflowExecution.query<number>(
+          "billingPeriodNumber"
+        );
+      const totalChargedAmount =
+        await subscriptionWorkflowExecution.query<number>("totalChargedAmount");
+
+      console.log("Workflow Id", subscriptionWorkflowExecution.workflowId);
+      console.log("Billing Period", billingPeriodNumber);
+      console.log("Total Charged Amount", totalChargedAmount);
 ```
+<!--SNIPEND-->
 
-You can set up Signals and Queries and Handlers for every field, or perhaps just one set for the entire Customer object, it depends on your needs.
-But these primitives can be flexibly rearranged (with specific inspiration from React Custom Hooks) to reduce boilerplate for your needs, [as we have documented](https://docs.temporal.io/typescript/workflows#signals-and-queries-design-patterns).
+In the above code, the `WorkflowHandle.query` method is used to request the `billingPeriodNumber` and `totalChargedAmount` from the running Workflow Execution without interrupting its execution. The results are then printed to the console.
 
-Knowledge check time - Write scripts with Temporal Clients for:
+## Using `condition` with timeouts
```

Comment:
Feels like this should definitely be part of the "Develop the Workflow" section?

This is a key aspect of getting the Workflow Entity behavior from TypeScript.

Seems odd to keep it at the end right before the conclusion.

Reply:
The "Develop a Workflow" section does not introduce signals yet, so it would be a fwd reference if I included that there, but thanks for the suggestion to move it! I'm going to move it to after Signals instead.